compiler/AST/iterator.cpp

```diff
@@ -862,6 +857,39 @@ static void replaceLocalWithFieldTemp(SymExpr*       se,
 // E.g. 'yield localvar' is converted to ic.value = ic.FNN_localvar.
 //
 
+static void replaceLocalUseOrDefWithFieldRef(SymExpr* se,
+                                             Symbol* classOrRecord,
+                                             std::vector<BaseAST*>& asts,
+                                             SymbolMap& local2field,
+                                             Vec<SymExpr*>& defSet,
+                                             Vec<SymExpr*>& useSet) {
+  if (useSet.set_in(se) || defSet.set_in(se)) {
+    // SymExpr is among those we are interested in: def or use of a live local.
+
+    // Get the corresponding field in the iterator class.
+    Symbol* field = local2field.get(se->symbol());
+
+    // Get the expression that sets or uses the symexpr.
+    CallExpr* call = toCallExpr(se->parentExpr);
+
+    if (call && call->isPrimitive(PRIM_ADDR_OF)) {
+
+      // Convert (addr of var) to (. _ic field).
+      // Note, GET_MEMBER is not valid on a ref field;
+      // in that event, GET_MEMBER_VALUE returns the ref.
+      if (field->isRef())
+        call->primitive = primitives[PRIM_GET_MEMBER_VALUE];
+      else
+        call->primitive = primitives[PRIM_GET_MEMBER];
```

Comment:
I think when debugging primitives -- and how they pop in and out of existence -- I actually grep the source code for the primitive name, and thus would fine a `new CallExpr` and `primitives[...]` just the same. That said, it's definitely magical when an AST node changes primitive in the middle of nowhere, and it would get in the way of the "traditional" mode of Chapel debugging.

Reply:
Yeah. I don't think we do this particular modification you're doing there too much. But I think this is a case where `CallExpr::setPrimitive` with a hook for `break-on-something-id` would be interesting if this pattern was more prevalent.

Your workflow of grepping is something I do as well, but I don't think that would work efficiently for the primitives you have here as I am sure we have a ton of instances of those.

In any case, as I said before, I am not asking for change here. I don't think this is ideal, but you're just moving the code and reaching my ideal is non-trivial for this PR.

Resolving this conversation.